docs/GUIDE.MD

```diff
@@ -0,0 +1,45 @@
+# How To Write Test Harness Tests
+
+Refer to the Variable Cloud tests and Variable Local tests on how to write tests for both Cloud and Local bucketing.
+
+Tests are divided into separate features that are present in each SDK tested by the test harness. The tests are located under
+`harness/features/`. There are also helper methods located under `harness/helpers.ts` that can be used to create SDK clients, call certain
+features on the proxy server and return the response, etc.
+
+An important helper method is the `forEachSDK` helper method, which can be wrapped around any tests to run the tests for all SDKs listed under
+`harness/types/sdks.ts`.
+
+Another useful method is to wrap tests with a `describeCapability` function, passing in a capability listed under `harness/types/capabilities.ts`. 
+This is used to segment different SDKs with different tests that are present on some SDKs but not all, e.g. Python not having a local bucketing capability.
+
+Lastly, we use `nock` to mock out server responses sent from the SDK. This allows to stub out certain methods on different URLs and return an object which can
+then we can use `jest` to expect the response from the proxy server.
+
+```
+e.g. This stubs out any calls from the SDK for the /variables endpoint with the specific variable key "var_key" and returns a variable object.
+
+Note: we can also granularly match on query params, this example matches on if the "enableEdgeDb" query param is set.
+scope
+    .post(`/client/${testClient.clientId}/v1/variables/var_key`, (body) => body.user_id === 'user1')
+    .matchHeader('Content-Type', 'application/json')
+    .matchHeader('authorization', testClient.sdkKey)
+    .query((queryObj) => {
+        return !!queryObj.enableEdgeDB
+    })
+    .reply(200, {
+        key: 'var_key',
+        value: 'value',
+        defaultValue: 'default_value',
+        type: 'String',
+        isDefaulted: false
+    })
+```
+
+We can then use our helper methods to call the `variable` method on our proxy servers: 
+```
+const variableResponse = (await testClient.callVariable(
+    { user_id: 'user1' },
+    'var_key',
+    'default_value'
+)).json()
+```
```

Comment:
maybe add something about how unsatisfied nock interceptors will cause a test case to fail at the end?

also might be good to have a little architecture overview of how all the different pieces fit together

Reply:
I added a bit on nock above, but what were you thinking about architecture overview that wouldn't already be covered in the README?

Reply:
well a general overview of how it actually works, maybe that block diagram we have showing the different pieces.

it would also be helpful to explain in general terms how all those pieces of the diagram actually function, eg. how do the proxies get run during tests? how does the harness communicate with them? how do the proxies know how to communicate back to the harness? how does the nock scope get hooked up with the mock server, and how does that server get started?

Reply:
@ajwootto ok i added a blurb in the guide with the accompanying diagram